### DISCUSSION TITLE
Discord OAuth2 does not return state to redirectUrl

### DISCUSSION THREAD

**Discussion by razhterize** [pocketbase/pocketbase#3783] [Q&A] [answered]
Hello, I'm currently using Pocketbase with Flutter to make CRUD app. I already enabled and tested OAuth2 and it is working perfectly fine on `Android` and `Windows.` Now here's the problem, after `Discord OAuth2` page, it should redirect to `https://<pocketbase>/api/oauth2-redirect?code=<code>&state=<state>`, but it does not give any state parameter.
Here's my `Discord OAuth2` implementation

EDIT:
I forgot to mention that only `Web` build and debug does not return `state`

```
Future<void> loginWithDiscord() async {
    emit(state.copyWith(loginStatus: LoginStatus.processing));
    if (pb.authStore.isValid) {
      var model = await pb.collection("discord_auth").authRefresh();
      emit(state.copyWith(loginStatus: LoginStatus.success, pb: pb, authModel: model));
      return;
    }
    try {
      await pb.collection("discord_auth").authWithOAuth2(
        "discord",
        (url) async {
          await launchUrl(url);
        },
        scopes: ['identify', 'guilds'],
      ).then((value) {
        if (pb.authStore.isValid) {
          authRefresh();
        }
      });
    } catch (e) {
      emit(state.copyWith(loginStatus: LoginStatus.failed));
    }
  }
```

**Comment by ganigeorgiev** [comment, marked as answer]
The "all-in-one" OAuthh2 flow relies on a realtime subscription to work and this is not supported on Flutter Web out of the box.

For Flutter Web you can use a custom browser based `http.Client` implementation similar to `fetch_client`. For example how to enable it you can check https://github.com/pocketbase/dart-sdk#limitations.

**Reply by razhterize** [reply]
> For Flutter Web you can use a custom browser based `http.Client `implementation similar to `fetch_client`

alright, I will try it first
